src/style.rs

```diff
@@ -96,7 +96,7 @@ impl Style {
     /// assert_eq!(colored.style().contains(Styles::Italic), true);
     /// assert_eq!(colored.style().contains(Styles::Dimmed), false);
     /// ```
-    pub fn contains(&self, style: Styles) -> bool {
+    pub fn contains(self, style: Styles) -> bool {
```

Comment:
are you sure about this change ? I don't think that consuming the style for a query is wise

Reply:
I was also confused by this initially, but according to rust this is actually more performant because the size of passing around `Style` is smaller than the size of passing around a reference to it:

take a look at this output:
```
warning: this argument (1 byte) is passed by reference, but would be more efficient if passed by value (limit: 8 byte)
  --> src/style.rs:99:21
   |
99 |     pub fn contains(&self, style: Styles) -> bool {
   |                     ^^^^^ help: consider passing by value instead: `self`
   |
   = note: `#[warn(clippy::trivially_copy_pass_by_ref)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#trivially_copy_pass_by_ref

```

Reply:
interesting ! TIL ! :)